src/pages/Markets/components/LpGrid.tsx

```diff
@@ -46,6 +58,52 @@ export const LpGrid: React.FC<{
     [assetIds, limit, selectedChainId],
   )
 
+  const sortedAssetIds = useMemo(() => {
+    if (!sortBy) return filteredAssetIds
+
+    return filteredAssetIds.sort((a, b) => {
+      const dataKey = marketDataBySortKey[sortBy]
+      const firstAssetId = orderBy === OrderOptionsKeys.ASCENDING ? a : b
+      const secondAssetId = orderBy === OrderOptionsKeys.ASCENDING ? b : a
+
+      if (dataKey === 'apy') {
+        const firstAssetOpportunityData = selectStakingOpportunityByFilter(store.getState(), {
```

Comment:
Performance improvement: extract our state above so we have a single reference to it:

```ts
const state = store.getState()
```

Reply:
Done in [1ac600a](https://github.com/shapeshift/web/pull/8006/commits/1ac600aeb95da5729ef78865c837fd0bcec0f777)